### DISCUSSION TITLE
Get MediaGroup

### DISCUSSION THREAD

**Discussion by RmzzzCode** [nutgram/nutgram#785] [Q&A]
Could you please tell me how to receive and process a media_group? I don’t see a specific method for this. When I send a batch of photos, they arrive as separate updates. How can I handle a specific number of photos in a batch? Are there any options available for this?

**Comment by Lukasss93** [comment]
That's how the Telegram Bot API works...
To handle this, you will need to check the `media_group_id` property of the message type on each update.

**Reply by fenestron** [reply]
When a user forwards a message with a `MediaGroup`, I get `media_group_id` `!=` `null`, but the `photo` array contains different `PhotoSize` only for the last photo. No other updates are received during forwarding. How can I get all photos for the received `media_group_id`?

**Reply by fenestron** [reply]
I also noticed that if you remove the sender name when forwarding, then a separate message is sent for each attachment. But if you forward with the sender name, only one message is sent.

**Comment by Kromvel-X** [comment]
I have encountered this problem and came up with this, not very concise solution for myself, you can explore it here: https://github.com/nutgram/nutgram/discussions/676#discussioncomment-10626443
